### PR TITLE
platform dependent modules

### LINKED ISSUE #7: Not fully compatible with yarn package manager
```bash
yarn add exiftool-vendored
node index.js
OK
```

```bash
yarn add anything-else
node index.js
Error: Cannot find module 'exiftool-vendored.pl'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/srv/nodejs/test-exiftool/node_modules/exiftool-vendored/dist/ExifTool.js:17:20)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
```

`index.js` includes only import of `exiftool-vendored` module

Works ok only on clean install or if `exiftool-vendored.pl` explicitly added as a package dependency.

This does not cause much inconvenience in a case of `exiftool-vendored` is a direct dependency. But it is not so if there are intermediate dependencies.
For example `module-a -> module-b -> exiftool-vendored`. Whenever a `module-a` adds a new dependency, `mobule-b` will throw an error.

### PR DESCRIPTION
**Pull request by amokrushin**
closes #7

### CODE REVIEW

File: postinstall.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+var process = require("process")
+var isWin32 = process.platform === "win32"
+
+try {
+  require("exiftool-vendored." + (isWin32 ? "exe" : "pl"))
```
Comment (mceachen) [inline postinstall.js]: I'd like this code to not duplicated in both ExifTool and postinstall.

**Comment by mceachen** [comment]
Looks good, thanks! I'll DRY up the require line and merge.

**Comment by amokrushin** [comment]
That's good, that looks good.
BTW, keep in mind, that line
https://github.com/mceachen/exiftool-vendored.js/blob/3b0a2c0eb2ec22caaac9be051a8ea48e59e7c666/src/ExifTool.ts#L19
is unreachable, because previous require will throw an error if module not exist.

**Comment by amokrushin** [comment]
The continuation of the idea how to solve problem with exiftool path.
Tests will not pass until exiftool-vendored.exe and exiftool-vendored.pl not published.
I've tested it locally on ubuntu and windows 10, temporary replacing `optionalDependencies` with
```json
 "optionalDependencies": {
    "exiftool-vendored.exe": "https://github.com/amokrushin/exiftool-vendored.exe#exiftool-path",
    "exiftool-vendored.pl": "https://github.com/amokrushin/exiftool-vendored.pl#exiftool-path"
  }
```
all tests excluding version check passed.

File: src/ExifTool.ts [hunk 2]
```diff
@@ -5,19 +5,12 @@ import { TagsTask } from "./TagsTask"
...
 export { Tags } from "./Tags"
 export { ExifDate, ExifTime, ExifDateTime, ExifTimeZoneOffset } from "./DateTime"
 
-const isWin32 = _process.platform === "win32"
-const exiftoolPath = require(`exiftool-vendored.${isWin32 ? "exe" : "pl"}`)
-
-if (!_fs.existsSync(exiftoolPath)) {
-  throw new Error(`Vendored ExifTool does not exist at ${exiftoolPath}`)
-}
+const exiftoolPath = _child_process.execSync("npm run -s exiftool-path").toString()
```
Comment (mceachen) [inline src/ExifTool.ts]: I really don't want to sync-exec `npm` in the middle of the require flow. I'll rework this before it goes into master.

**Comment by amokrushin** [comment]
Anyway, thank you for great module!

**Comment by amokrushin** [comment]
p.s. Previous idea with `"exiftool": "./bin/exiftool"` do not looks suitable. Because perl version of exiftool relies on cwd to find ifself lib, strange behavior, but it is what it is. On the other side, windows version of npm do not creates symlink, but *.cmd wrapper, and most of the exiftool tests are failed in this case.

**Comment by mceachen** [comment]
Turns out the `postinstall` target is abysmal for this use--there isn't a way to discriminate between local development (where we'd need to compile) and normal tarball consumption of the module (where the compiled source is in the tarball already). I've dropped the postinstall validation script.

Thanks again for your help and ideas.

**Pull request closed without merging**